sway-lsp/src/handlers/request.rs

```diff
@@ -0,0 +1,398 @@
+//! This module is responsible for implementing handlers for Language Server
+//! Protocol. This module specifically handles requests.
+
+use crate::{
+    capabilities,
+    global_state::{GlobalState, GlobalStateSnapshot},
+    lsp_ext,
+    utils::debug,
+};
+use forc_tracing::{init_tracing_subscriber, TracingSubscriberOptions, TracingWriterMode};
+use lsp_types::{
+    CodeLens, CompletionResponse, DocumentFormattingParams, DocumentSymbolResponse,
+    InitializeResult, InlayHint, InlayHintParams, PrepareRenameResponse, RenameParams,
+    SemanticTokensParams, SemanticTokensResult, TextDocumentIdentifier, Url, WorkspaceEdit,
+};
+use std::{
+    fs::File,
+    io::Write,
+    path::{Path, PathBuf},
+};
+use sway_types::{Ident, Spanned};
+use tower_lsp::jsonrpc::Result;
+use tracing::metadata::LevelFilter;
+
+pub(crate) fn handle_initialize(
+    state: &GlobalState,
```

Comment:
Same comment here, if we can avoid having to pass `&GlobalState` to every function, I think it's cleaner.

In the functions below where `snap` is passed as a parameter, that could be replaced by calling `self.state.snapshot()` directly.

We could even make the dynamic `sessions` inside of `GlobalState` private so  handlers are forced to call `snapshot()`.

Reply:
I agree that it would make more sense for these methods to take self and have them implemented on `GlobalState` directly as the `LanguageServer` trait is implemented for this type. That means that essentially the LSP event loop implemented in tower is operating on `GlobalState`. 

This won't be the case is the follow up PR for the event loop so I would like to keep these as they are for now. We can re look if this is still and issue when the eventloop PR is ready.